src/main/java/org/traccar/protocol/Gt06ProtocolDecoder.java

```diff
@@ -794,22 +794,34 @@ private Object decodeBasic(Channel channel, SocketAddress remoteAddress, ByteBuf
 
             position.set(Position.KEY_TYPE, type);
 
+            if (type == MSG_GPS_LBS_2 && "NT20".equals(model)) {
+                buf.readUnsignedByte(); // location source type
+                buf.skipBytes(8); // imei
+
+                DateBuilder dateBuilder = new DateBuilder((TimeZone) deviceSession.get(DeviceSession.KEY_TIMEZONE))
+                        .setDate(buf.readUnsignedByte(), buf.readUnsignedByte(), buf.readUnsignedByte())
+                        .setTime(buf.readUnsignedByte(), buf.readUnsignedByte(), buf.readUnsignedByte());
+
+                position.setDeviceTime(dateBuilder.getDate());
+            }
+
             if (hasGps(type)) {
                 decodeGps(position, buf, false, deviceSession.get(DeviceSession.KEY_TIMEZONE));
             } else {
                 getLastLocation(position, null);
             }
 
             if (hasLbs(type) && buf.readableBytes() > 6) {
-                boolean hasLength = hasStatus(type)
+                boolean hasLength = (hasStatus(type)
+                        || ("NT20".equals(model) && type == MSG_GPS_LBS_2))
```

Comment:
Seems like this part should inside `hasStatus` check.

Reply:
HasStatus only receives message type

Reply:
Currently yes.